lib/ansible/galaxy/collection/concrete_artifact_manager.py

```diff
@@ -508,17 +512,22 @@ def _download_file(url, b_path, expected_hash, validate_certs, token=None, timeo
     return b_file_path
 
 
-def _consume_file(read_from, write_to=None):
+def _consume_file(read_from, write_to=None, length=None):
```

Comment:
I recommend going for kw-only args for new stuff, when the arg list starts growning.
```suggestion
def _consume_file(read_from, write_to=None, *, length=None):
```